### ISSUE TITLE
Can't handle arrow function inside static property which inside class extends Mixin

### ISSUE THREAD

**Issue by xiaohanzhang** [pangloss/vim-javascript#716]
In following example, the closing } of bar will become red.
```
class Foo extends Mixin({a: 1, b: 2}) {
  static foo = someFunc({
     bar: (a, b) => {
       const { args } = a;
     }
  });
}
```

**Comment by bounceme** [comment]
Is this one bug?

**Comment by amadeus** [comment]
Yeah, this is a bug, and probably a non-trivial one to fix. The issue is not so much the arrow function, it's due the opening `{` not being matched properly as a class block...

**Comment by amadeus** [comment]
I have a fix ready in #717

**Comment by xiaohanzhang** [comment]
the fix works great! Thanks!
